dhcpv6/option_bootfileparam.go

```diff
@@ -0,0 +1,76 @@
+package dhcpv6
+
+import (
+	"bytes"
+	"encoding/binary"
```

Comment:
This library has been using `github.com/u-root/u-root/pkg/uio` pretty extensively for parsing/writing, it has a couple useful features esp. around error handling that you end up reimplementing in this PR. I think it would be a good idea to also use it here

Reply:
Created an issue: https://github.com/insomniacslk/dhcp/issues/341

Reply:
Well. I don't understand why `uio` was used over `binary`. The error handling did not become easier (I still have to write `if`s in the same places). Moreover with `uio` errors became ["anonymous"](https://github.com/u-root/u-root/blob/801a65c9a22bcc5dba4d02208b0c94b074dc9bae/pkg/uio/buffer.go#L100), so they cannot be compared/checked in any way in future (except substring-search which is not reliable). But I agree that code consistency is important. So OK and thanks :)